### PR TITLE
Gauss potentials

### PR DESCRIPTION
**Pull request by kjs73**
This adds Gaussian shaped potentials.
Includes also an initializer list constructor for array.

### CODE REVIEW

**Comment by js850** [comment]
I'm having a look now, but can you add some documentation, It's not quite clear from the name what a gaussian potential is.  I think I understand, but I had to look at the code

File: source/pele/array.h [hunk 1]
```diff
@@ -108,6 +109,17 @@ class Array
           _size(_memory->size())
 
     {}
+    
+    /**
+     * construct an array from an initializer list
+     */
+    Array(std::initializer_list<dtype> data)
```
Comment (js850) [inline source/pele/array.h]: cool :-)

**Comment by js850** [comment]
Can SumGaussianPot be rewritten using CombinedPotential?  It seems to be doing the same thing.

https://github.com/pele-python/pele/blob/master/source/pele/combine_potentials.h#L15

You can probably even do it on the cython level following BLJCut or something similar

https://github.com/pele-python/pele/blob/master/pele/potentials/_lj_cpp.pyx#L124

File: source/pele/sumgaussianpot.h [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+          m_cov_diag(cov_diag.copy()),
+          m_gauss_prefactor(-1),
+          m_bdim(mean.size())
+    {
+        assert(m_mean.size() == m_bdim);
+        assert(m_cov_diag.size() == m_bdim);
+        for (size_t i = 0; i < m_bdim; ++i) {
+            m_diag_icov.push_back(1. / m_cov_diag[i]);
+        }
+        m_diag_icov.swap(m_diag_icov);
```
Comment (js850) [inline source/pele/sumgaussianpot.h]: what is the purpose of this line?  swapping a vector with itself?

**Comment by js850** [comment]
Other than the issues I raised, and of course the failing test, I would be happy to merge.

**Comment by js850** [comment]
btw, I think the failing test is because you modified setup_with_cmake.py but not setup.py

File: source/pele/sumgaussianpot.h (hunk 2 above)
Comment (kjs73) [inline source/pele/sumgaussianpot.h]: This has been replaced by shrink_to_fit now: http://www.cplusplus.com/reference/vector/vector/shrink_to_fit/

**Comment by kjs73** [comment]
Thank you! Setup is updated now.

**Comment by kjs73** [comment]
OK, now it is rewritten using CombinedPotential. Some documentation is added.

**Comment by js850** [comment]
hmm, the current travis failure seems unrelated to this PR.  I'll try to have a look at it in the next few days.  Maybe some package we depend on changed the way they do things?

**Comment by kjs73** [comment]
It seems that there is a problem with munkres now. Maybe it has changed with the new version? But their example still looks the same as what is done in the code: https://pypi.python.org/pypi/munkres/

**Comment by js850** [comment]
munkres hasn't been updated since 2014-12-03.

I assume that some networkx function returns a set now instead of a list.  But I haven't had time to check yet.

**Comment by js850** [comment]
i fixed the failing test in PR #133 

If you merge master into this PR the tests should pass.

**Comment by kjs73** [comment]
Thank you!

**Comment by js850** [comment]
btw, can you now delete the c++ class SumGaussianPot?

**Comment by kjs73** [comment]
The cpp class SumGaussianPot can be deleted. Will also rename the corresponding header file.

**Comment by js850** [comment]
Hi Julian, I finally fixed the test that was failing , which means this PR can be merged now if you're happy with it

**Comment by kjs73** [comment]
Hi Jake, Thank you! Just merged with your changes to master. I think it should work fine now.

**Pull request merged**